docs/usage.rst

```diff
@@ -0,0 +1,209 @@
+.. Specify pandas format output in cells
+.. jupyter-execute::
+    :hide-code:
+    :hide-output:
+
+    import pandas as pd
+
+
+    def series_to_html(self):
+        df = self.to_frame()
+        df.columns = ['']
+        return df._repr_html_()
+
+
+    setattr(pd.Series, '_repr_html_', series_to_html)
+    pd.set_option('display.max_rows', 6)
+
+.. Specify version for storing and loading objects to YAML
+.. jupyter-execute::
+    :hide-code:
+
+    __version__ = '1.0.0'
+
+
+Usage
+=====
+
+The basic idea of :mod:`audinterface` is
+to provide easy and standardized interfaces
+to apply a machine learning model,
+or other digital signal processing algorithms
+to audio files.
+The only prerequisite is
+the algorithm provides a callable
+that takes at least the signal
+as a :class:`numpy.ndarray`
+and the sampling rate as input.
+
+The interface can then apply the algorithm
+on a list of files,
+a folder,
+or an index conform to the audformat_ database specification.
+Results are always returned containing a `segmented index`_.
+In the following we load three files from the emodb_ database
+and define a list of files,
+a folder,
+and an index.
+
+.. jupyter-execute::
+
+    import audb
+    import os
+
+    media = ['wav/03a01Fa.wav', 'wav/03a01Nc.wav', 'wav/03a01Wa.wav']
+    db = audb.load('emodb', version='1.2.0', media=media, verbose=False)
+
+    files = list(db.files)
+    folder = os.path.dirname(files[0])
+    index = db['emotion'].index
+
+
+Create a processing interface
+-----------------------------
+
+Let's assume we want to calculate the root mean square (RMS)
+value in dB.
+We first define the function
+and create an interface for it using :class:`audinterface.Process`.
+
+.. jupyter-execute::
+
+    import audinterface
+    import numpy as np
+
+    def rms(signal, sampling_rate):
+        return 20 * np.log10(np.sqrt(np.mean(signal ** 2)))
+
+    interface = audinterface.Process(process_func=rms)
+
+The following three commands
+apply the algorithm
+and all return the same result
+as a :class:`pandas.Series`.
+
+.. jupyter-execute::
+
+    y = interface.process_files(files)
+    y = interface.process_folder(folder)
+    y = interface.process_index(index)
+    y
+
+
+Create a feature extractor
+--------------------------
+
+When using an algorithm as feature extractor,
```

Comment:
Maybe we should be a little bit more generic here. E.g.: When the result of the processing function has multiple dimensions it is recommended to use :class:`audinterface.Feature`, which returns a :class:`pandas.DataFrame`
with named dimensions.

Reply:
Changed accordingly.